jobs/sql/incremental/deleteQuery.sql

```diff
@@ -0,0 +1,5 @@
+delete t from ${tableName}_${partitionNum} t
+inner join patient_last_update_date d on t.patient_id = d.patient_id
+where d.partition_num = ${partitionNum}
+and d.last_updated >= @previousWatermark
+and d.last_updated <= @newWatermark
```

Comment:
This delete query is deleting out of the partition any rows (based on the partition-modified patient_id column in the target table), any rows that have been updated since the last watermark and the newly identified (future) watermark.  In order for us to reuse this for any of our jobs, we need any that want to do incremental loading to have a patient_id column containing ${partitionNum}-patient_id as a value.

Reply:
So, in this case, if any patient change as been recorded since the last watermark, this should delete all encounters from for that patient from target all_encounters table?  (Makes sense to me, just want to make sure I'm understanding).

Reply:
Correct.  Since all we track are changes to patient, the approach is to first delete all rows related to any patients that have changed since the last watermark (so that we an subsequently load/insert new data related to those changed patients).